### ISSUE TITLE
[BUG] Cannot specify build directory in bdist_wheel

### ISSUE THREAD

**Issue by rossburton** [pypa/setuptools#4732]
### setuptools version

72.1.0

### Python version

3.9

### OS

Linux

### Description

To ensure clean builds, we want to build python modules with a fresh build directory:

```
$ python3 setup.py build -b /a/fresh/directory
```

However, we can't then use this build tree with `bdist_wheel` as there's no option to specify where the build tree is when using `--skip-build`:

```
$ python3 ./setup.py bdist_wheel -b /a/fresh/directory --skip-build
running bdist_wheel
installing to /a/fresh/directory
running install
running install_lib
warning: install_lib: 'build/lib' does not exist -- no Python modules to install
```

Yes I know invoking setup.py is deprecated but there's no way to pass `-j` to `build`...

### Expected behavior

`bdist_wheel` has an option to tell is where the build tree is.

### How to Reproduce

$ python3 setup.py build -b /a/fresh/directory
$ python3 setup.py bdist_wheel -b /a/fresh/directory

### Output

```console
warning: install_lib: 'build/lib' does not exist -- no Python modules to install
```

**Comment by abravalheri** [comment]
You can try setting up the relative fields in `setup.cfg` for the `[build]` and `[bdist_wheel]` sections.

Please note that using `setup.py` as a CLI tool is deprecated (since the implementation of PEP 517, `setup.py` is only seen as a configuration file, not as a CLI), so it no longer receives the same level of support of other functionalities (or feature requests).

**Comment by rossburton** [comment]
My goal is to do this for _all_ packages we build, I just discovered `DIST_EXTRA_CONFIG` and it appears that setting it to a file like this does the right thing:

```
[build]
build_base = /my/build/path
```

This appears to be the only way to set the build directory when using a pep517 build tool?

**Comment by abravalheri** [comment]
Yes, in the current state of implementation of setuptools that should work.

**Comment by rossburton** [comment]
Gentle prod that this is still an issue.

We're doing large-scale building of python packages (linux distro building) and want to be _sure_ that the build is clean and doesn't have artifacts from previous builds.

 As we're building distro packages, we don't use build isolation.  I tried something like this:

```
$ python3 -mbuild --verbose --no-isolation -C--global-option=build -C--global-option=--build-base=/tmp/FOFOFO
```

But this resulted in it building several times, the '* Getting build dependencies for sdist...' part for example did a full build and then '* Building sdist...' did it again.

**Comment by rossburton** [comment]
Actually this relates to another issue where I want to build a wheel with `-j` but there's no simply way to pass `-j` to `build_ext` via `bdist_wheel`.

**Comment by rossburton** [comment]
Finally, the `DIST_EXTRA_CONFIG` trick doesn't work anymore.
